src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -199,3 +201,14 @@ internals ints = interfaceConcept $ \s ->
                  if s `elem` ints
                  then Internal
                  else Unused
+
+combinationalGate :: Eq a => Expr a -> a -> CircuitConcept a
```

Comment:
We could change this to: 
```haskell
combinationalComplexGate :: Eq a => Expr a -> Transition a -> CircuitConcept a
```
And allow a user to pass `rise x` or `fall x` into this, so they are effectively setting `set`/`reset` themselves. If they do 
```haskell 
combinationalComplexGate (And (Var a) (Var b)) (rise x) <> combinationalComplexGate (Or (Not (Var a)) (Not (Var b))) (fall x)
```
As an example, and then also implement `complexGate` with a set and reset function.